web/src/containers/ExecutiveSummary.js

```diff
@@ -8,85 +9,112 @@ import { expandActivitySection } from '../actions/activities';
 import Dollars from '../components/Dollars';
 import { Section, Subsection } from '../components/Section';
 import { t } from '../i18n';
+
+import { jumpTo } from '../actions/navigation';
 import { selectApdYears } from '../reducers/apd.selectors';
-import { selectBudgetExecutiveSummary } from '../reducers/budget.selectors';
+import {
+  selectBudgetExecutiveSummary,
+  selectBudgetGrandTotal
+} from '../reducers/budget.selectors';
+
+class ExecutiveSummary extends PureComponent {
```

Comment:
If you have the time/inclination, I'd love to know more about why you changed this from a function-based component to a class-based pure component! Just my curiousity, though. :)

Reply:
I changed it to a class because it has an instance method now, and functional components can't do that.  I used `PureComponent` because they render more efficiently.  `PureComponent` implements the React `shouldComponentUpdate` lifecycle method and does a shallow comparison of the current state and props to the incoming state and props.  If they match, it does not re-render.  The standard `Component` class, meanwhile, doesn't do any comparison so it will re-render every time the state or props change (or the parent re-renders!), even if the new and old are internally identical.  Those words don't make a lot of sense to me, so here's an example:

```js
const props1 = {
  name: 'Alice',
  age: 37
};

const props2 = {
  name: 'Alice',
  age: 37
};
```

In a standard `Component`, switching from props1 to props2 would cause a re-render because they are different objects.  A `PureComponent`, however, would shallowly compare the values in the two objects, see that nothing changed, and _*not*_ re-render.